AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/GraphQLModelBased/GraphQLModelBasedTests.swift

```diff
@@ -294,6 +294,63 @@ class GraphQLModelBasedTests: XCTestCase {
         wait(for: [completeInvoked], timeout: TestCommonConstants.networkTimeout)
     }
 
+    func testUpdatePostWithPredicate() {
+        let uuid = UUID().uuidString
+        let testMethodName = String("\(#function)".dropLast(2))
+        let title = testMethodName + "Title"
+        guard let createdPost = createPost(id: uuid, title: title) else {
+            XCTFail("Failed to ensure at least one Post to be retrieved on the listQuery")
+            return
+        }
+        let updatedTitle = title + "Updated"
+        let updatedPost = Post(id: uuid,
+                               title: updatedTitle,
+                               content: createdPost.content,
+                               createdAt: createdPost.createdAt)
+        let post = Post.keys
+        var predicate = post.title == "incorrectPreviousTitle"
+        let completedWithErrorInvoked = expectation(description: "request completed with error")
+        _ = Amplify.API.mutate(of: updatedPost, where: predicate, type: .update, listener: { event in
+            switch event {
+            case .completed(let data):
+                switch data {
+                case .success(let post):
```

Comment:
Looks like we have one test which tests the .failure case, and the .success case.. i think it would be better to separate this to two separate test cases.

Reply:
updating